src/components/med_field_row.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from "react";
+import { med_field } from "../app_config";
+interface Props {
+  med_field: typeof med_field[0];
+  nr_indicators: number;
+  update_med_field_filter(arr: string[]): void;
+  update_clicked_med_field(str: string): void;
+  clicked_med_field: string;
+}
+function MED_FIELD_ROW(props: Props) {
+  const {
+    med_field,
+    nr_indicators = 10,
+    update_med_field_filter,
+    update_clicked_med_field,
+    clicked_med_field,
+  } = props;
+  const class_checked =
+    med_field.react_key === clicked_med_field ? "checked" : "";
+
+  const handle_med_field_click = () => {
+    update_clicked_med_field(med_field.react_key);
+    update_med_field_filter(med_field.key);
+  };
+
+  useEffect(() => {
+    if (med_field.react_key === clicked_med_field) {
+      update_med_field_filter(med_field.key);
+    }
+  }, [med_field, clicked_med_field, update_med_field_filter]);
```

Comment:
Do we need this? Looks like the click handler updates both.

Reply:
You are right, it's redundant.